node/full_client_test.go

```diff
@@ -970,7 +970,7 @@ func TestMempool2Nodes(t *testing.T) {
 		require.NoError(node2.Stop())
 	}()
 
-	time.Sleep(1 * time.Second)
+	time.Sleep(3 * time.Second)
```

Comment:
no! haha

what is the condition that we are waiting on? how can we check it explicitly in a retry so that we aren't relying on flaky bare sleeps.

Reply:
@gupadhyaya could you please answer this? What's the best alternative solution to this?

Reply:
@Manav-Aggarwal its unclear how to change this arbitrary timeout to sync-based. let's create an issue for further investigation and unblock this PR.

Reply:
Created an issue for this: https://github.com/rollkit/rollkit/issues/1130